### PR TITLE
Add crate and module to glossary.

### LINKED ISSUE #1015: Glossary: add "crate" and "module"
Please add "crate" and "module" to the glossary. As someone coming from Python, I'm a bit unsure of what Rust terminology to use instead of Python's "package".

### PR DESCRIPTION
**Pull request by ehuss**
Closes #1015

### CODE REVIEW

**Comment by tlyu** [comment]
These look good to me, for what it's worth. I wonder if it would help to mention that modules might or might not be in separate files, and that crates can consist of multiple source files. (Arguably, mentioning these in the glossary might be redundant with the full description in the body of the reference. Also, Rust's terminology here might be sufficiently idiosyncratic that a small amount of clarification in the glossary would help.)

File: src/glossary.md [hunk 1]
```diff
@@ -130,6 +140,14 @@ This is not affected by applied type arguments. `struct Foo` is considered local
 `Vec<Foo>` is not. `LocalType<ForeignType>` is local. Type aliases do not
 affect locality.
 
+### Module
+
+A module is a container for zero or more [items]. Modules are organized in a
+tree, starting from an unnamed module at the root called the crate root or the
+root module. [Paths] may be used to refer to items outside of a module, which
```
Comment (matthewjasper) [inline src/glossary.md:147]: "outside of the module that defines them"? "from other modules"?

Comment (ehuss) [reply src/glossary.md]: Yea, the sentence was quite awkward. I went with your second suggestion, thanks!

**Review by matthewjasper — APPROVED** [review]

**Comment by matthewjasper** [comment]
Thanks

**Pull request merged**